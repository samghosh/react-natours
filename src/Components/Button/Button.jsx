import React from 'react';
import { string } from 'prop-types';

// Importing the Compiled CSS of the Button Component
import './Button.css';

const Button = ({ buttonText, buttonType }) => {
    return (
        <>
            <a href="#" className={ `btn ${ buttonType }` }>
<<<<<<< HEAD
                {buttonText}
=======
                <span data-testid="button-text">{buttonText}</span>
>>>>>>> parent of 9432730... Project Updated to suit a different portfolio
            </a>
        </>
    )
};

Button.defaultProps = {
    buttonType: 'btn-white'
}

Button.propTypes = {
    buttonText: string.isRequired,
    buttonType: string
}

export default Button;