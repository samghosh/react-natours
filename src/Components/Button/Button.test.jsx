import React from 'react';
import Button from './Button';
import { render } from '@testing-library/react';

it('matches snapshot', () => {
    const { asFragment } = render(<Button buttonText='Button Text' />);
    expect(asFragment()).toMatchSnapshot();
});

it('should render props passed exactly', () => {
    const { getByTestId } = render(<Button buttonText='Hello World' />);
    expect(getByTestId('button-text').textContent).toBe('Hello World');
});