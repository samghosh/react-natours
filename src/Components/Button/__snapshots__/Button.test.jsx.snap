// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="btn btn-white"
    href="#"
  >
<<<<<<< HEAD
    Button Text
=======
    <span
      data-testid="button-text"
    >
      Button Text
    </span>
>>>>>>> parent of 9432730... Project Updated to suit a different portfolio
  </a>
</DocumentFragment>
`;
